### PR TITLE
enhancement(humio_logs sink): add support for sub-milliseconds timestamps

### LINKED ISSUE #10203: Take advantage of `timestamp.nanos` when possible in the `humio_logs` sink
Humio offers [support for sub-millsecond timestamps](https://library.humio.com/stable/docs/parsers/parsing-timestamps/#sub-millisecond-precision-of-timestamps) when available. We should use this field when Vector's own timestamp has a granularity larger than milliseconds.

### PR DESCRIPTION
**Pull request by prognant**
closes #10203

Slight modification to allow sub-ms parts of timestamps to be added to event sent by the splunk-hec sink and derivatives.
It is disabled for splunk, but comes with the right config for the humio-logs sinks (as per https://library.humio.com/stable/docs/parsers/parsing-timestamps/#sub-millisecond-precision-of-timestamps).

Tested against https://cloud.community.humio.com/services/collector/event.

### CODE REVIEW

File: website/cue/reference/components/sinks/splunk_hec_logs.cue [hunk 1]
```diff
@@ -141,6 +141,18 @@ components: sinks: splunk_hec_logs: {
 				examples: ["${SPLUNK_HEC_TOKEN}", "A94A8FE5CCB19BA61C4C08"]
 			}
 		}
+		timestamp_nanos_key: {
```
Comment (jszwedko) [inline website/cue/reference/components/sinks/splunk_hec_logs.cue:144]: I think `@timestamp.nanos` is something special, specific to Humio. I'd suggest we leave this configuration option off of the `splunk_hec_logs` sink.

**Review by jszwedko — COMMENTED** [review]
Generally looks good. I think that this notion of a nanosecond timestamp field is something specific to Humio though so I hesitate to have configuration for it for the Splunk HEC sink generally.

If it is possible, it'd be nice to _only_ have this configurable / set for the `humio_logs` sink and not the `splunk_hec_logs` sink. I realize that could require some rework though since the current implementation of the `humio_logs` sink is a simple wrapper around the `splunk_hec_logs` sink. At the least, I think we should:

* Leave the configuration out of the docs for the `splunk_hec_logs` sink, even though it will exist
* Leave a comment that we added this field to the `splunk_hec_logs` sink out of convenience, but that it isn't actually relevant. If the `humio_logs` sink diverges more we could consider separating it more than a simple wrapper.

**Review by jszwedko — COMMENTED** [review]
Looks good!

File: src/sinks/splunk_hec/logs/sink.rs [hunk 2]
```diff
@@ -124,6 +127,7 @@ pub fn process_log(
     index: Option<&Template>,
     host_key: &str,
     indexed_fields: &[String],
+    timestamp_nanos_key: Option<&String>,
```
Comment (bruceg) [inline src/sinks/splunk_hec/logs/sink.rs:130]: Can this be `Option<&str>`? `&String` is usually discouraged as being pointless.

**Review by bruceg — COMMENTED** [review]
This code looks good to me, but this is missing updates to the reference for the new configuration field.

**Comment by prognant** [comment]
> This code looks good to me, but this is missing updates to the reference for the new configuration field.
The issue is that the option is really useless for the `splunk_hec` and is only useful (and comes with the relevant default) for the `humio_logs`.  I [initially added a chunk of doc](https://github.com/vectordotdev/vector/pull/10216/commits/d6528af5be7d5d03df6c10276420c681d459e2f6#diff-ca6e75cfb084f5c7e80f543f191a8c0478676b0e2f7685e39b3ca1813430dd63), but in the end it seemed a bit confusing

File: src/sinks/splunk_hec/logs/sink.rs (hunk 2 above)
Comment (prognant) [reply src/sinks/splunk_hec/logs/sink.rs]: Yes, I updated the PR accordingly.

**Review by jszwedko — APPROVED** [review]

**Pull request merged**